### PR TITLE
Bump to Kubernetes 1.25.3 and Update all dependencies + Go 1.19.3

### PR DESCRIPTION
**Pull request by Dentrax**
Signed-off-by: Furkan <furkan.turkal@trendyol.com>

As the title already mentions the purpose, this PR bumps every dependency to latest version and removes deprecated / old packages.

* Dependencies are quite old and packages have massive change so far.
* The main k/k repo [was updated to Go 1.19.3](https://github.com/kubernetes/kubernetes/pull/112287) for the k8s v1.25.3 release.

- [X] Bump Go 1.19.3 (Deprecates https://github.com/kubernetes/node-problem-detector/pull/693)
- [X] Bump to Kubernetes 1.25.3
- [X] Remove deprecated libraries
- - [X] ~`k8s.io/utils/inotify`~: Use fsnotify
- - [X] ~`k8s.io/kubernetes/pkg/ssh`~: ([Deprecated](https://github.com/kubernetes/kubernetes/compare/v1.21.14...v1.22.0) at v1.22.0) Adopted from latest stable version (with tests)
- - [X] ~`k8s.io/heapster`~: Use clientcmd
- - [X] ~`cadvisor/tail`~: ([Deprecated](https://github.com/google/cadvisor/compare/v0.36.0...v0.37.0) at v0.37.0 with [this commit](https://github.com/google/cadvisor/commit/af89abfc29bb9f5e169820c3bac381420b3924e9)) Adopted from latest stable version (with tests)
- - [X] Use new `sigs.k8s.io/boskos`
- - [X] Use new `ginkgo/v2` [Migrated to V2](https://onsi.github.io/ginkgo/MIGRATING_TO_V2)

I was able to get success to build on my macOS. We will see about Unit and E2E tests.

* This PR requires a minor version bump once it gets merged. (We couldn't guarantee backward compatibility and stability like as-is. So we have to trust tests.)
* It would be nice to add Dependabot.

### CODE REVIEW

**Comment by Dentrax** [comment]
Kind ping @vteratipally @xueweiz, it would be nice to keep review time window short to avoid possible conflicts. 🤞 Do you have a bandwidth for this?

**Comment by Dentrax** [comment]
Hey @vteratipally @xueweiz, looks like my all tests are happy except `pull-npd-e2e-node`. That's because it complains about `Master not detected. Is the cluster up?` So eventually ginkgo timeouts and throws the following: (I'm not sure if it's a known issue or because of my PR.)

```
   Timed out after 60.001s.
  Expected success, but got an error:
      <*errors.errorString | 0xc00045f1f0>: {
          s: "node condition \"TestCondition\" not found",
      }
      node condition "TestCondition" not found
  In [It] at: test/e2e_node/node_problem_detector_linux.go:419 
```

---

Other than these, I think I'm ready for the review. bf22a971c9f11008c7be4c82aea8eeb64de0907f is my final commit. 🤞

**Comment by Dentrax** [comment]
kind ping @vteratipally

**Comment by ssoriche** [comment]
Can this be merged? There are many CVEs related to packages used within this project and this PR resolves them all.

**Comment by ssoriche** [comment]
Unfortunately I could not get this MR working in our environment.

Without specifying the `- --apiserver-override=https://$(KUBERNETES_SERVICE_HOST):$(KUBERNETES_SERVICE_PORT)` I get the following error message in the node-problem-detector log:

```
panic: host must be a URL or a host:port pair: "://"

goroutine 1 [running]:
k8s.io/client-go/kubernetes.NewForConfigOrDie(...)
        k8s.io/client-go@v0.25.3/kubernetes/clientset.go:649
k8s.io/node-problem-detector/pkg/exporters/k8sexporter/problemclient.NewClientOrDie(0xc00012e0d0)
        k8s.io/node-problem-detector/pkg/exporters/k8sexporter/problemclient/problem_client.go:91 +0x4d8
k8s.io/node-problem-detector/pkg/exporters/k8sexporter.NewExporterOrDie(0xc00012e0d0)
        k8s.io/node-problem-detector/pkg/exporters/k8sexporter/k8s_exporter.go:52 +0x3d
main.npdMain(0xc00012e0d0, 0x1?)
        k8s.io/node-problem-detector/cmd/nodeproblemdetector/node_problem_detector.go:61 +0xc5
main.npdInteractive(0xc000194100?)
        k8s.io/node-problem-detector/cmd/nodeproblemdetector/node_problem_detector.go:38 +0x72
main.main()
        k8s.io/node-problem-detector/cmd/nodeproblemdetector/node_problem_detector_linux.go:29 +0x1d9
```

When specifying the option with `- --apiserver-override=https://$(KUBERNETES_SERVICE_HOST):$(KUBERNETES_SERVICE_PORT)`, I get the following:

```
I0112 14:30:09.435757       1 log_watchers.go:40] Use log watcher of plugin "journald"
I0112 14:30:09.436065       1 k8s_exporter.go:54] Waiting for kube-apiserver to be ready (timeout 5m0s)...
E0112 14:30:09.637689       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:14.645988       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:19.644243       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:24.646602       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:29.642583       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:34.647849       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:39.642793       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:44.646642       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:49.643475       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:54.646908       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:30:59.643809       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:31:04.645495       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
E0112 14:31:09.642638       1 k8s_exporter.go:111] Can't get node object: Get "https://100.64.0.1:443/api/v1/nodes/ip-10-20-171-236.us-west-2.compute.internal": x509: certificate signed by unknown authority
```

I verified that the ServiceAccount exists and is available in the pod.

I had a look through the code and started reverting some of the changes to restore functionality, but ultimately I have to abandon this path.

**Comment by Dentrax** [comment]
Hey @ssoriche, thank you for testing this. I can add you as a collaborator to my fork if you want. Do you want to send the fix? I've been thinking to update dependencies again since passed a while from last bump.

File: pkg/util/k8s_ssh/k8s_ssh_test.go [hunk 1]
```diff
@@ -0,0 +1,361 @@
+/*
```
Comment (vteratipally) [inline pkg/util/k8s_ssh/k8s_ssh_test.go:1]: This particular method is not encourage across kubernetes, it is better to switch to running `ssh` command

**Pull request closed without merging**